pkg/controller/operators/catalog/operator.go

```diff
@@ -1437,113 +1430,35 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 	}
 
 	ensurer := newStepEnsurer(kubeclient, crclient, dynamicClient)
+	b := newBuilder(kubeclient, dynamicClient, o.csvProvidedAPIsIndexer)
 
 	for i, step := range plan.Status.Plan {
-		switch step.Status {
-		case v1alpha1.StepStatusPresent, v1alpha1.StepStatusCreated:
+		doStep := true
+		s, err := b.step(step.Status, step.Resource, step.Resource.Name)
```

Comment:
I think this should take the whole step as a single argument.

Reply:
Agreed. I renamed the function to `create` so we have `create(step)` for clarity as well.